egasub/ega/services/__init__.py

```diff
@@ -117,14 +150,29 @@ def submit_run(ctx, run):
     }
     
     r = requests.post(url,data=json.dumps(run.to_dict()), headers=headers)
-    print r.text
+    #print r.text
     r_data = json.loads(r.text)
     
-    if r_data['header']['code'] != 200:
-        print r_data['header']['userMessage']
-        return r_data['response']['result'][0]['id']
+    if r_data['header']['code'] == "200":
+        run.id = r_data['response']['result'][0]['id']
     else:
-        return r_data['header']['userMessage']
+        #TODO
+        raise Exception(r_data['header']['userMessage'])
+    
+
+
+def submit_submission(ctx,submission):
+    url = "%ssubmissions/%s?action=SUBMIT" % (EGA_SUB_URL_PROD,ctx.obj['SUBMISSION']['id'])
+    
+    headers = {
+        'Content-Type': 'application/json',
+        'X-Token' : ctx.obj['SUBMISSION']['sessionToken']
+    }
+       
+    r = requests.put(url,data=json.dumps(submission.submission_subset.to_dict()), headers=headers)
```

Comment:
This `submission.submission_subset.to_dict()` should be just `submission.to_dict()`